src/libpriv/rpmostree-scripts.c

```diff
@@ -0,0 +1,402 @@
+  { "%postun", 0,
+    RPMTAG_POSTUN, RPMTAG_POSTUNPROG, RPMTAG_POSTUNFLAGS },
+  { "%triggerun", RPMSENSE_TRIGGERUN,
+    RPMTAG_TRIGGERUN, 0, 0 },
+  { "%triggerpostun", RPMSENSE_TRIGGERPOSTUN,
+    RPMTAG_TRIGGERPOSTUN, 0, 0 },
+};
+#endif
+
+static const KnownRpmScriptKind posttrans_scripts[] = {
+  /* For now, we treat %post as equivalent to %posttrans */
+  { "%post", 0,
+    RPMTAG_POSTIN, RPMTAG_POSTINPROG, RPMTAG_POSTINFLAGS },
+  { "%posttrans", 0,
+    RPMTAG_POSTTRANS, RPMTAG_POSTTRANSPROG, RPMTAG_POSTTRANSFLAGS },
+};
+
+static const KnownRpmScriptKind unsupported_scripts[] = {
+  { "%prein", 0,
+    RPMTAG_PREIN, RPMTAG_PREINPROG, RPMTAG_PREINFLAGS },
+  { "%pretrans", 0,
+    RPMTAG_PRETRANS, RPMTAG_PRETRANSPROG, RPMTAG_PRETRANSFLAGS },
+  { "%triggerprein", RPMSENSE_TRIGGERPREIN,
+    RPMTAG_TRIGGERPREIN, 0, 0 },
+  { "%triggerin", RPMSENSE_TRIGGERIN,
+    RPMTAG_TRIGGERIN, 0, 0 },
+  { "%verify", 0,
+    RPMTAG_VERIFYSCRIPT, RPMTAG_VERIFYSCRIPTPROG, RPMTAG_VERIFYSCRIPTFLAGS},
+};
+
+static void
+child_setup_fchdir (gpointer user_data)
+{
+  int fd = GPOINTER_TO_INT (user_data);
+  if (fchdir (fd) < 0)
+    err (1, "fchdir");
+}
+
+static void
+add_const_args (GPtrArray *argv_array, ...)
+{
+  va_list args;
+  char *arg;
+
+  va_start (args, argv_array);
+  while ((arg = va_arg (args, char *)))
+    g_ptr_array_add (argv_array, arg);
+  va_end (args);
+}
+
+static void
+fusermount_cleanup (const char *mountpoint)
+{
+  g_autoptr(GError) tmp_error = NULL;
+  const char *fusermount_argv[] = { "fusermount", "-u", mountpoint, NULL};
+  int estatus;
+
+  if (!g_spawn_sync (NULL, (char**)fusermount_argv, NULL, G_SPAWN_SEARCH_PATH,
+                     NULL, NULL, NULL, NULL, &estatus, &tmp_error))
+    {
+      g_prefix_error (&tmp_error, "Executing fusermount: ");
+      goto out;
+    }
+  if (!g_spawn_check_exit_status (estatus, &tmp_error))
+    {
+      g_prefix_error (&tmp_error, "Executing fusermount: ");
+      goto out;
+    }
+
+ out:
+  /* We don't want a failure to unmount to be fatal, so all we do here
+   * is log.  Though in practice what we *really* want is for the
+   * fusermount to be in the bwrap namespace, and hence tied by the
+   * kernel to the lifecycle of the container.  This would require
+   * special casing for somehow doing FUSE mounts in bwrap.  Which
+   * would be hard because NO_NEW_PRIVS turns off the setuid bits for
+   * fuse.
+   */
+  if (tmp_error)
+    sd_journal_print (LOG_WARNING, "%s", tmp_error->message);
+}
+
+static RpmOstreeScriptAction
+lookup_script_action (HifPackage *package,
+                      GHashTable *ignored_scripts,
+                      const char *scriptdesc)
+{
+  const char *pkg_script = glnx_strjoina (hif_package_get_name (package), ".", scriptdesc+1);
+  const struct RpmOstreePackageScriptHandler *handler = rpmostree_script_gperf_lookup (pkg_script, strlen (pkg_script));
+  if (ignored_scripts && g_hash_table_contains (ignored_scripts, pkg_script))
```

Comment:
Shouldn't we look up the package name here instead of `pkg_script`? Or maybe both actually, that way we can do e.g. `--ignore-script gdb` to whitelist all, or just `--ignore-script gdb.prein` to whitelist just that one.